packages/webapp/src/components/Form/Unit/useUnit.jsx

```diff
@@ -233,33 +229,34 @@ const useUnit = ({
       onBlur(e);
     }
   };
+
+  // set visible value when hidden input is updated from parent component
+  // (this happens when the component shows a result of a calculation)
   useEffect(() => {
-    if (databaseUnit && hookFormUnit) {
-      setVisibleInputValue(
-        hookFormValue > 0 || hookFormValue === 0
-          ? roundToTwoDecimal(convert(hookFormValue).from(databaseUnit).to(hookFormUnit))
-          : '',
-      );
+    if ((noValue(visibleInputValue) && noValue(hookFormValue)) || !databaseUnit || !hookFormUnit) {
+      return;
+    }
+
+    const newValue = roundToTwoDecimal(convert(hookFormValue).from(databaseUnit).to(hookFormUnit));
+    if (newValue !== visibleInputValue) {
+      setVisibleInputValue(hookFormValue > 0 || hookFormValue === 0 ? newValue : '');
     }
   }, [hookFormValue]);
 
   const getMax = useCallback(() => {
     return hookFormUnit ? convert(max).from(hookFormUnit).to(databaseUnit) : max;
   }, [hookFormUnit, max, databaseUnit]);
 
-  const getOnChangeUnitOption = (onChange) => {
-    return (e) => {
-      // function from react-hook-form Controller's render
-      onChange(e);
+  const onChangeUnit = (e) => {
+    hookFormSetValue(displayUnitName, e);
```

Comment:
https://github.com/react-hook-form/react-hook-form/blob/52925875e7a5ab41601e45c3209f6c5bda0099ff/src/logic/createFormControl.ts#L655
onChange from controller does more than just setting value. 
The safe way is to use useController hook. 
but hookFormSetValue is fine since there is no validation on unit